libkbfs/folder_update_prepper.go

```diff
@@ -0,0 +1,1223 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"fmt"
+
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/kbfs/kbfsblock"
+	"github.com/keybase/kbfs/tlf"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// folderUpdatePrepper is a helper struct for preparing blocks and MD
+// updates before they get synced to the backend servers.  It can be
+// used for a single update or for a batch of updates (e.g. conflict
+// resolution).
+type folderUpdatePrepper struct {
+	config       Config
```

Comment:
Oh god, that looks horrific to me.  I know you guys are into it, but I can't see any benefit to it for this type. Can you explain why this is useful for this type?